exercism/annalyns-infiltration/LincolnInitialSolution.cs

```diff
@@ -0,0 +1,59 @@
+using System;
+
+static class QuestLogic
+{
+    public static bool CanFastAttack(bool knightIsAwake)
+    {
+        if(knightIsAwake){
+            return false;
+        }
+        else{
+            return true;
+        }
+    }
+
+    public static bool CanSpy(bool knightIsAwake, bool archerIsAwake, bool prisonerIsAwake)
+    {
+        if(knightIsAwake || archerIsAwake || prisonerIsAwake){
+            return true;
+        }
+        else{
+            return false;
+        }
+    }
+
+    public static bool CanSignalPrisoner(bool archerIsAwake, bool prisonerIsAwake)
+    {
+        if (archerIsAwake){
+            return false;
+        }
+        else{
+            if(prisonerIsAwake){
+                return true;
+            }
+            else{
+                return false;
+            }
+        }
+    }
+
+    public static bool CanFreePrisoner(bool knightIsAwake, bool archerIsAwake, bool prisonerIsAwake, bool petDogIsPresent)
+    {
+        if(petDogIsPresent){
+            if(archerIsAwake == false){
```

Comment:
Even leaving aside the suggestions above, you shouldn't need to compare booleans to boolean literals. The expression within an `if` statement merely needs to evaluate to `true` or `false`; it doesn't have to be _compared_ to `true` or `false`, because you can always get there through the boolean expression or its negation.